modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SDeclaration.java

```diff
@@ -37,13 +37,13 @@
  */
 public final class SDeclaration extends AStatement {
 
-    private final String type;
+    private DType type;
```

Comment:
Can you please add a comment here explaining why this is mutable and what are plans are moving forward here?

Reply:
Changed this to final instead. Will add mutable DResolvedType nodes where necessary as the conversions take place. These should be removed with the creation of a second tree.